nncf/common/batchnorm_adaptation.py

```diff
@@ -0,0 +1,107 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Optional
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from nncf.api.compression import ModelType
+from nncf.common.initialization import NNCFDataLoader
+import nncf.common.factory as factory
+
+
+class BatchnormAdaptationAlgorithmImpl(ABC):
```

Comment:
`impl` referce to the implementation, not the interface. Should probably call this `IBatchnormAdaptationRunner`.